### DISCUSSION TITLE
Shallow_Networks_I notebook. Question 5

### DISCUSSION THREAD

**Discussion by jamalromero** [udlbook/udlbook#228] [Q&A] [answered]
in [Shallow_Networks_I.ipynb ](https://github.com/udlbook/udlbook/blob/main/Notebooks/Chap03/3_1_Shallow_Networks_I.ipynb) there's a section of TODOs just before _Least squares loss_ section. Question 5 says:
```
# 5. Change the parameters so that there are only two "joints" (including outside the range of the plot)
# There are actually three ways to do this. See if you can figure them all out

```
I think there are more than 3 ways to achieve this. One of the solutions is to set:
 $\phi_1=0$ or $\phi_2=0$ or $\phi_3=0$
 
But we can achieve the same result if we set $\theta_{10} = \theta_{20}$ and $\theta_{11} = \theta_{21}$ simultaneously.

Curious to know if there's a closed form formulae on how to control the number of the desired joints we would like to have in the graph taking into account all the parameters.

Appologies to post this queston here. I would appreciate if you know any user group discussion available where to post relevamt questions like this.

Thanks!

**Comment by udlbook** [comment, marked as answer]
Hi!  

The three ways to make this happen are:

(i) To set phi1, phi2, or phi3 to zero (as you correctly identify)
(ii) To set theta11, theta21, or theta22 to zero.  This will mean that one of the lines in figures 3.3a-c will be horizontal and so it won't cross zero and the ReLU function won't make a kink in it.
(iii) To make two of the lines in figures 3.3a-c cross zero in the same place.  You identified one way to do this, but there are others.

I don't think there is a closed form formula for the number of joints (or the number regions between them is probably the more interesting quantity).  

No problem to ask questions... I'll definitely answer them all, but don't expect fast responses!

There isn't a user discussion group as far as I know, but you are welcome to set one up, or if you find something similar for another book that seems to work well, I don't mind setting it up.  Need advice about what would be most effective though!

**Reply by jamalromero** [reply]
Hi,
Thank you very much for the clarification! 
On point (iii), I think I found all the cases. The reasoning is as follows:
Calculating the $x$ coordinates of the three joints:

$θ_{10} + θ_{11}x = 0 \implies x_1 = \frac{-θ_{10}}{θ_{11}}$

$θ_{20} + θ_{21}x = 0 \implies x_2 = \frac{-θ_{20}}{θ_{21}}$

$θ_{30} + θ_{31}x = 0 \implies x_3 = \frac{-θ_{30}}{θ_{31}}$

Then if $x_1 \neq x_2 \neq x_3$, there will be three joints. If any two are equal, there will be two joints and if all three are equal there will be one or no joint. So :

If $\frac{θ_{10}}{θ_{11}} = \frac{θ_{20}}{θ_{21}}$

or if $\frac{θ_{10}}{θ_{11}} = \frac{θ_{30}}{θ_{31}}$

or if $\frac{θ_{20}}{θ_{21}} = \frac{θ_{30}}{θ_{31}}$

The graph will have only two joints

Thanks
